### PR TITLE
(Feature) Better error handling / token unlocking

### LINKED ISSUE #118: Better error handling
Better error handling. Perhaps highlight the textfield, and show the error just below it.

### LINKED ISSUE #206: Improve token unlocking workflow
Design: https://zpl.io/Vk1YkMZ

### PR DESCRIPTION
**Pull request by gabitoesmiapodo**
Closes #118 
Closes #206

**Notes:** 

- **Do not merge this PR** before #152 is approved and merged.
- "Max bidding price"'s tooltip's text is missing.
- There's no "token was approved (but it wasn't before)" status, so I can't make the last status message work for the token approval process. Not Approved / Is Approving status should work.

It should look like these:

Screen Shot 2021-05-15 at 13 33 05
Screen Shot 2021-05-15 at 13 33 15
Screen Shot 2021-05-15 at 13 33 27
Screen Shot 2021-05-15 at 13 33 47
Screen Shot 2021-05-15 at 13 34 10
Screen Shot 2021-05-15 at 13 51 34

May-15-2021 13-44-10
May-15-2021 13-46-34
May-15-2021 13-50-09

### CODE REVIEW

**Comment by josojo** [comment]
Really nice! this improves the ux sooo much! Thanks.

Checked many cases and it works great. On xdai and rinkeby.

2 improvements could be applied:
image
Here the user has done nothing wrong but gets already a red message about the order cancellation. It's frightening. Given that we show the information in the details, I think we can remove it here and just show it in the order confirmation as a reminder.
image
Here, the distance between the inputs and the orange button is quite huge. Just a proposal? What do you think about pulling it closer together?

**Comment by gabitoesmiapodo** [comment]
@josojo 

Maybe something like this could do it:

- I can change the `!` icon to a calendar icon (we are warning the user about a date, after all), and I can also change the color to white or something else to make it less threatening.
- And maybe we can increase the fields' height to use some more space? In the image I made them 20px taller, that could probably work and it's nothing complicated to do.

23e- GnosisAuction - Auction v2

**Comment by josojo** [comment]
I talked to @cmagan 
and we definitively like this version the best:
image

Could you implement it? Then we would test this PR throughly.

**Comment by josojo** [comment]
I just realized that the unlock buttton is very small and hardly visible/recognizeable
image
the design showed a different approach. Are you planning to implement this?

**Comment by josojo** [comment]
I also realized that the new auction details are always in the foreground:
image

File: src/components/auction/OrderPlacement/index.tsx [hunk 1]
```diff
@@ -324,23 +267,59 @@ const OrderPlacement: React.FC<OrderPlacementProps> = (props) => {
...
   const balanceString = React.useMemo(() => {
-    return account
-      ? chainId !== chainIdFromWeb3
-        ? 'Switch network'
-        : `${biddingTokenBalance?.toSignificant(6) || '0'} ${getTokenDisplay(
-            biddingToken,
-            chainId,
-          )}`
-      : 'Connect your wallet'
-  }, [account, biddingToken, biddingTokenBalance, chainId, chainIdFromWeb3])
```
Comment (josojo) [inline src/components/auction/OrderPlacement/index.tsx:343]: Why did you choose to delete this logic?

**Review by josojo — APPROVED** [review]
Really good! love the new desgin! smooth and slick

**Comment by josojo** [comment]
I can not get past this error:
Visit auction:
input some amount
keep the suggested price
invert the price

Then the parsing of the price no longer works :( and we get a weird error
image

**Pull request merged**